### PR TITLE
bump minimal python version to 3.10

### PR DESCRIPTION
**Pull request by dariuszd21**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

Fixes: #177

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v5
         with:
-          python-version: '3.11'
+          python-version: '3.10'
```
Comment (dariuszd21) [inline .github/workflows/tests.yaml:22]: Should we run linting against newest or oldest interpreter?

File: tox.ini [hunk 2]
```diff
@@ -43,15 +43,15 @@ extras = dev
...
 description =
     unit: Run unit tests with pytest
     integration: Run integration tests with pytest
 labels =
-    py3.{8,10,11}: tests
-    unit-py3.{8,10,11}: unit-tests
-    integration-py3.{8,10,11}: integration-tests
+    py3.{10,12}: tests
+    unit-py3.{10,12}: unit-tests
+    integration-py3.{10,12}: integration-tests
```
Comment (dariuszd21) [inline tox.ini:54]: Should we run here also against `3.11` ?

File: tox.ini [hunk 3]
```diff
@@ -3,7 +3,7 @@ env_list =  # Environments to run when called with no parameters.
     format-{black,ruff,codespell}
     pre-commit
     lint-{black,ruff,mypy,pyright,shellcheck,codespell,docs,yaml}
-    unit-py3.{8,10,11}
+    unit-py3.{10,12}
```
Comment (dariuszd21) [inline tox.ini:6]: Should we also run it against 3.11?

**Review by mr-cal — APPROVED** [review]
Looks good! 

I don't have strong opinions regarding your questions about which python versions to test on and will defer to @lengau.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (lengau) [reply .github/workflows/tests.yaml:22]: It doesn't really matter. In fact we could probably remove the `setup-python` step from linting and just use the system python if we wanted.

File: tox.ini (hunk 3 above)
Comment (lengau) [reply tox.ini:6]: `{oldest,newest}` is fine for the default environments.

File: tox.ini (hunk 2 above)
Comment (lengau) [reply tox.ini:54]: Here we should include all the versions.

**Review by lengau — APPROVED** [review]

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (dariuszd21) [reply .github/workflows/tests.yaml]: I removed `Setup Python` action from linting. Let's see how it ends

File: tox.ini (hunk 2 above)
Comment (dariuszd21) [reply tox.ini]: Changed to include also `3.11`

File: tox.ini (hunk 3 above)
Comment (dariuszd21) [reply tox.ini:6]: So I left it as is. Thank you!

**Pull request merged**